doc/source/complex-plugin.rst

```diff
@@ -41,11 +41,30 @@ between plugins).  Once the automagics have been constructed, the plugin can be
  * the configuration path within the context for the plugin
  * any callback to determine progress in lengthy operations
  * any file consumers for files created during running of the plugin
+ * a file handler class for files created during the running of the plugin
 
 With the constructed plugin, it can either be run by calling its
 :py:meth:`~volatility.framework.interfaces.plugins.PluginInterface.run` method, or any other known method can
 be invoked on it.
 
+Writing out files
+-----------------
+The :py:class:`~volatility.framework.interfaces.plugins.PluginInterface` contains a FileHandler attribute, that should be
+used as a means of writing out files by passing them to the `consume_file` method.  When producing a file, construct an object
+handler using the :py:obj:`~volatility.framework.interfaces.plugins.PluginInterface.FileHandler` class.
+
+..code-block:: python
+
+    output = self.FileHandler(filename = 'newfile.out')
+    output.data.write(data_for_file)
```

Comment:
Why is this extra `data` layer here? I think this should look more like `open(); write(); close()`

Reply:
That's from our `FileInterface`, which has a prefered name, and data.  I could override a filelike object with the preferred name, but I went with having a data attribute which contained the file like object.  This is absolutely our opportunity to change it though.  We'd need to bump some numbers because it would be a framework change if we did though (although I think we do that for this change anyway?)